src/Cache/Signature/FixerSignature.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of PHP CS Fixer.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *     Dariusz Rumiński <dariusz.ruminski@gmail.com>
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace PhpCsFixer\Cache\Signature;
+
+use PhpCsFixer\Fixer\FixerInterface;
+
+final class FixerSignature
+{
+    private string $name;
+    private string $contentHash;
+
+    /**
+     * @var array<string, mixed>|bool
+     */
+    private $config;
+
+    private function __construct()
+    {
+    }
+
+    /**
+     * @param array<string, mixed>|bool $config
+     */
+    public static function fromInstance(FixerInterface $fixer, $config): self
+    {
+        $signature = new self();
+        $signature->name = $fixer->getName();
+        $signature->contentHash = md5(file_get_contents((new \ReflectionClass($fixer))->getFileName()));
+        $signature->config = \is_array($config) ? self::makeJsonEncodable($config) : $config;
+
+        return $signature;
+    }
+
+    /**
+     * @param array<string, mixed>|bool $config
+     */
+    public static function fromRawValues(string $name, string $contentHash, $config): self
+    {
+        $signature = new self();
+        $signature->name = $name;
+        $signature->contentHash = $contentHash;
+        $signature->config = \is_array($config) ? self::makeJsonEncodable($config) : $config;
+
+        return $signature;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function getContentHash(): string
```

Comment:
it took me 10 minutes to understand what `content` means here

Reply:
I don't like redundancy and I believe namespace + class name + method name is a full context to get the intention. So **Fixer**Signature + get**Content**Hash shows it's about fixer's content (code). How would you want it to be named?